src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -1066,7 +1066,7 @@ const intptr_t OneBit     =  1; // only right_most bit set in a word
 // get a word with the n.th or the right-most or left-most n bits set
 // (note: #define used only so that they can be used in enum constant definitions)
 #define nth_bit(n)        (((n) >= BitsPerWord) ? 0 : (OneBit << (n)))
-#define right_n_bits(n)   (nth_bit(n) - 1)
+#define right_n_bits(n)   ((uintptr_t) nth_bit(n) - 1)
```

Comment:
This changes the return type of right_n_bits, which could break existing code.  If we need an unsigned version, I think it should have a different name.

Reply:
And this is at best a partial fix: `OneBit << 63` overflows.